nomad/structs/bitmap.go

```diff
@@ -17,6 +17,32 @@ func NewBitmap(size int) (Bitmap, error) {
 	return Bitmap(b), nil
 }
 
+// Copy returns a copy of the Bitmap
+func (b Bitmap) Copy() (Bitmap, error) {
+	if b == nil {
+		return nil, fmt.Errorf("can't copy nil Bitmap")
+	}
+
+	nb, err := NewBitmap(b.Size())
+	if err != nil {
+		return nil, err
+	}
+
+	s := b.Size()
+	for i := uint(0); i < s; i++ {
```

Comment:
I would copy full bytes at a time not bits. You have access to the underlying byte array, and there is no need to add a conditional set.